File: Code_c#_Lab_1/Code_c#_Lab_1/NumberCalculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Code_c__Lab_1
{
    class NumberCalculator
    {
        private int threadId;
        private int increment;
        private volatile bool stopped = false;
        private CountdownEvent startLatch;

        public NumberCalculator(int threadId, int increment, CountdownEvent startLatch)
        {
            this.threadId = threadId;
            this.increment = increment;
            this.startLatch = startLatch;
        }

        public void SetStopped(bool stopped)
        {
            this.stopped = stopped;
        }

        public void Run()
        {
            startLatch.Wait(); // Очікуємо сигналу запуску

            long totalSum = 0;
            long termsCount = 0;

            while (!stopped)
            {
                totalSum += increment;
                termsCount++;
            }

            Console.WriteLine($"Thread ID: {threadId}, Sum: {totalSum}, Terms: {termsCount}, Step: {increment}");
        }
    }
}

File: Code_java_Lab_1/untitled/src/NumberCalculator.java
import java.util.concurrent.CountDownLatch;

class NumberCalculator extends Thread {
    int threadId;
    int increment;
    volatile boolean stopped = false;

    public NumberCalculator(int threadId, int increment) {
        this.threadId = threadId;
        this.increment = increment;
    }

    public void setStopped(boolean stopped) {
        this.stopped = stopped;
    }

    @Override
    public void run() {

        long totalSum = 0;
        long termsCount = 0;

        while (!stopped) {
            totalSum += increment;
            termsCount++;
        }

        System.out.println("Thread ID: " + threadId +
                ", Sum: " + totalSum +
                ", Terms: " + termsCount +
                ", Step: " + increment);
    }
}
